### PR TITLE
API: Refactor `argmax` and `argmin`

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This PR refactors the current implementation of `argmax` and `argmin`. It removes "only 0.0 fill value" and "all non-fill values greater/less than the fill value" constraints. It also makes the core of the function jittable. 

First https://github.com/pydata/sparse/pull/613 needs to be merged.

### CODE REVIEW

**Comment by hameerabbasi** [comment]
I think the comments from the GitHub bot are valid in this PR and should be addressed.

File: sparse/_coo/common.py [hunk 1]
```diff
@@ -1144,3 +1081,127 @@
...
+    # Compute max/min arguments
+    result_indices, result_data = _compute_minmax_args(
+        x.coords.copy(),
+        x.data.copy(),
+        reduce_size=x.shape[0],
+        fill_value=x.fill_value,
+        max_mode_flag=max_mode_flag,
+    )
+
+    from .core import COO
```
[the remark this answers is not in the document]
Comment (mtsokol) [reply sparse/_coo/common.py:1191]: It's a common pattern in `sparse`.

**Comment by mtsokol** [comment]
> I think the comments from the GitHub bot are valid in this PR and should be addressed.

Hi @hameerabbasi,
Now PR is ready from my side.

**Comment by hameerabbasi** [comment]
Thanks for your awesome work on this @mtsokol.

**Pull request merged**